File: rotate-matrix.js
var input = [ [29, 62, 8], [12, 60, 31], [99, 79, 12] ];

function rotateMatrix(array) {

  var output = [];

  for (var i = 0; i < array.length; i++) {
    output.push([]);
    console.log(output);
  }

  for (var i = 0; i < array.length; i++) {
    for (var j = 0; j < array[0].length; j++) {
      output[j].unshift(array[i][j]);
      console.log(i, j);
      console.log(output);
    }
  }

  return output;
}
console.log(rotateMatrix(input));

File: rotate-matrix.swift
var array: [[Int]] = [ [29, 62, 8], [12, 60, 31], [99, 79, 12] ]

func rotateMatrix(input: [[Int]]) -> [[Int]] {

  var output:[[Int]] = []

  for _ in input {
    output.append([])
  }

  for t in 0..<input.count {
    for i in 0..<input[0].count {
      output[t].insert(input[i][t], atIndex: 0)
    }
  }

  return output
}

print(rotateMatrix(array))
